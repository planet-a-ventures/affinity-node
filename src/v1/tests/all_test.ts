import { assertInstanceOf } from '@std/assert/mod.ts'
import { assertSnapshot } from '@std/testing/snapshot.ts'
import { afterEach, beforeEach, describe, it } from '@std/testing/bdd.ts'
import * as path from '@std/path/mod.ts'

import axios from 'axios'
import MockAdapter from 'axios-mock-adapter'
import Affinity from '../index.ts'
import _ from 'https://esm.sh/v135/axios@1.7.1/denonext/unsafe/helpers/toFormData.js'

const __dirname = path.dirname(path.fromFileUrl(import.meta.url))
const apiKey = Deno.env.get('API_KEY')
const isLiveRun = typeof apiKey !== 'undefined'

async function getRawFixture(filePath: string) {
    return await Deno.readTextFile(path.join(__dirname, 'fixtures', filePath))
}

describe('Affinity', () => {
    let mock: MockAdapter
    let affinity: Affinity

    beforeEach(() => {
        if (!isLiveRun) {
            mock = new MockAdapter(axios)
        }
        affinity = new Affinity(apiKey || 'api_key')
    })
    afterEach(() => {
        mock?.reset()
    })

    describe('whoami', () => {
        it('can be called', async (t) => {
            mock?.onGet('/auth/whoami').reply(
                200,
                await getRawFixture('whoami.raw.response.json'),
            )
            const res = await affinity.whoAmI()

            assertInstanceOf(res.grant.createdAt, Date)
            await assertSnapshot(t, res)
        })
    })

    describe('rate_limit', () => {
        it('can be called', async (t) => {
            mock?.onGet('/rate-limit').reply(
                200,
                await getRawFixture('rate_limit.raw.response.json'),
            )
            const res = await affinity.rateLimit()
            await assertSnapshot(t, res)
        })
    })
})
