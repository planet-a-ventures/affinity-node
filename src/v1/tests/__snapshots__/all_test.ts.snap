export const snapshot = {};

snapshot[`Affinity > whoami > can be called 1`] = `
{
  grant: {
    createdAt: 2000-01-01T00:00:00.000Z,
    scope: "external_api",
    type: "api_key",
  },
  tenant: {
    id: 12345,
    name: "Workspace Name",
    subdomain: "workspace",
  },
  user: {
    email: "first@last.com",
    firstName: "First",
    id: 123456789,
    lastName: "Last",
  },
}
`;

snapshot[`Affinity > rate_limit > can be called 1`] = `
{
  rate: {
    api_key_per_minute: {
      limit: 900,
      remaining: 878,
      reset: 10,
      used: 22,
    },
    org_monthly: {
      limit: Infinity,
      remaining: Infinity,
      reset: 2124845,
      used: 7,
    },
  },
}
`;
